crates/sdk/src/cuda/builder.rs

```diff
@@ -25,19 +26,38 @@ impl CudaProverBuilder {
     /// ```rust,no_run
     /// use sp1_sdk::ProverClient;
     ///
-    /// let prover = ProverClient::builder().cuda().with_moongate_endpoint("http://...").build();
+    /// let prover = ProverClient::builder().cuda().with_external_moongate_server("http://...").build();
     /// ```
     #[must_use]
-    pub fn with_moongate_endpoint(mut self, endpoint: &str) -> Self {
-        self.moongate_endpoint = Some(endpoint.to_string());
-        self
+    pub fn with_external_moongate_server(
```

Comment:
```suggestion
    pub fn server(
```

This is simpler, and easier to set up.